lib/perl/Genome/Sys/Lock.pm

```diff
@@ -97,6 +97,8 @@ sub lock_resource {
 
     $class->_cleanup_handler_check();
 
+    $RESOURCE_LOCK_SCOPE{$args{resource_lock}} = $args{scope};
```

Comment:
That is currently unsupported, but could be a feature add.  I'm not sure what the use case would be right now.